File: be/src/main/java/com/example/Campung/notification/dto/NotificationListResponse.java
package com.example.Campung.notification.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationListResponse {
    
    private List<NotificationResponse> notifications;
    private long unreadCount;
    private int totalPages;
    private long totalElements;
    private int currentPage;
    private int size;
}

File: android/campung/app/src/main/java/com/shinhan/campung/data/remote/response/NotificationListResponse.kt
package com.shinhan.campung.data.remote.response

import com.google.gson.annotations.SerializedName

data class NotificationListResponse(
    @SerializedName("notifications")
    val notifications: List<NotificationResponse>? = null,
    
    @SerializedName("unreadCount")
    val unreadCount: Long = 0,
    
    @SerializedName("totalPages")
    val totalPages: Int = 0,
    
    @SerializedName("totalElements")
    val totalElements: Long = 0,
    
    @SerializedName("currentPage")
    val currentPage: Int = 0,
    
    @SerializedName("size")
    val size: Int = 0
)